lib/blocks.php

```diff
@@ -220,7 +220,15 @@ function gutenberg_wpautop_block_content( $content ) {
  */
 function gutenberg_wpautop_insert_post_data( $data ) {
 	if ( ! empty( $data['post_content'] ) && gutenberg_content_has_blocks( $data['post_content'] ) ) {
-		$data['post_content'] = gutenberg_wpautop_block_content( $data['post_content'] );
+		// WP_REST_Posts_Controller slashes post data before inserting/updating
```

Comment:
We can expect this filter (`wp_insert_post_data `) will be called not in the context of a REST API save, but generally any save for post content. Do we need to account for this? Is it wasteful (or worse, breaking) for us to be unslashing in these other contexts?

Reply:
Good questions.

Is it breaking? My reasoning:
- A [comment](https://github.com/WordPress/WordPress/blob/4e320fe0087e8df83e6ba7f7765dbc7100c20067/wp-includes/post.php#L3324-L3325) in `wp_insert_post()`, upon creating `$data` from a bunch of fields, reads _Expected slashed_, referring to said fields.
- At the end, `wp_insert_post()` blindly unslashes all of `$data`, with no prior check for certain kinds of formatting in `$data`, which I took as another sign that it _is_ expected that `wp_insert_post` be passed slashed data.
- It would be odd if saving via the REST API would silently differ in such a crucial aspect.
- Assuming that all other actors (incl. third parties) play well and provide slashed data, there shouldn't be anything breaking about this, as that would imply that `$x !== wp_slash( wp_unslash( $x ) )`.

Is it wasteful? Plausibly; we'd have to measure this. We could bypass this for non-REST-API contexts, but from my first paragraph it seems like the better default to leave it on.